File: Python/Lesson_02/Task_01.py
# Task 1

# Напишите программу, которая принимает на вход вещественное число и показывает сумму его цифр.
# Пример:
# 0,56 -> 11

num = input("Enter number: ")
sum = 0

for i in str(num):
    if (i != '.') and (i != ','):
        sum += int(i)

print(sum)

File: Exceptions in programming/Lesson_02/src/Task_01.java
import java.util.InputMismatchException;
import java.util.Scanner;

public class Task_01 {

    public static float prompt() {
        float num = 0F;
        while (true) {
            Scanner scn = new Scanner(System.in);
            System.out.print("Enter double number please: ");
            try {
                num = scn.nextFloat();
                return num;
            } catch (InputMismatchException e) {
                System.out.println("You're wrong, try again please");
            } finally {
                System.out.println("Accepted");
            }
        }
    }

    public static void main(String[] args) {

        // 1. Реализуйте метод, который запрашивает у пользователя ввод дробного числа (типа float),
        //    и возвращает введенное значение. Ввод текста вместо числа не должно приводить к падению приложения,
        //    вместо этого, необходимо повторно запросить у пользователя ввод данных.

        System.out.println(prompt());
    }
}
